src/App.tsx

```diff
@@ -1,223 +1,223 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useState,
+  useCallback,
+  useMemo,
+} from 'react';
+import classNames from 'classnames';
 import { AuthContext } from './components/Auth/AuthContext';
+import {
+  getTodos, addTodo, deleteTodo, updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { NewTodoField } from './components/Auth/NewTodoField/NewTodoField';
+import { TodoList } from './components/Auth/TodoList';
+import { ErrorNotification } from './components/Auth/ErrorNotification';
+import { Status } from './types/Status';
+import { Filter } from './components/Auth/Filter';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [title, setTitle] = useState('');
+  const [error, setError] = useState('');
+  const [status, setStatus] = useState(Status.All);
+  const [isAdding, setIsAdding] = useState(false);
+  const [processings, setProcessings] = useState<number[]>([]);
+
+  const addProcessing = (id: number) => {
+    setProcessings(current => [...current, id]);
+  };
+
+  const removeProcessing = (idToRemove: number) => {
+    setProcessings(current => current.filter(id => id !== idToRemove));
+  };
+
+  const loadUserTodos = useCallback(() => {
+    if (!user) {
+      return;
+    }
 
-  useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    setError('');
+
+    try {
+      getTodos(user.id)
+        .then(setTodos);
+    } catch {
+      setError('Can not load todos');
     }
-  }, []);
+  }, [user]);
+
+  useEffect(() => {
+    loadUserTodos();
+  }, [user]);
+
+  const handleSubmit = useCallback(
+    async (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+
+      if (!title) {
+        setError('Title can\'t be empty');
+
+        return;
+      }
+
+      setIsAdding(true);
```

Comment:
You set isAdding here as true, but if program catch an error (for examle after set in title several spaces '   '), it will disable button forever. You should add isAdding false if program catch an error